user/homework.go

```diff
@@ -32,6 +33,7 @@ const (
 type Homework struct {
 	Items    []*Item
 	template string
+	width    int
```

Comment:
`maxTitleWidth` is being called 2 times and both times from within `Homework.Report`. I think we should calculate the max width at the top of the `Report` function, store it in a local variable, and use that value in `Item.Report` and `hw.heading`. The reason for `width` being a property of `Homework` in not obvious and adds a piece of state that must now be maintained.